src/SFA.DAS.Forecasting.Api/Startup.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using MediatR;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Authorization;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using SFA.DAS.Forecasting.Application.AccountProjection.Queries;
+using SFA.DAS.Forecasting.Application.AccountProjection.Services;
+using SFA.DAS.Forecasting.Data;
+using SFA.DAS.Forecasting.Data.Repository;
+using SFA.DAS.Forecasting.Domain.AccountProjection;
+using SFA.DAS.Forecasting.Domain.Configuration;
+using SFA.DAS.Forecasting.Domain.Validation;
+using SFA.DAS.Forecasting.Infrastructure.Configuration;
+
+namespace SFA.DAS.Forecasting.Api
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            var config = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json")
+                .AddEnvironmentVariables()
+                .AddAzureTableStorageConfiguration(
+                    configuration["ConfigurationStorageConnectionString"],
+                    configuration["AppName"].Split(","),
+                    configuration["Environment"],
+                    configuration["Version"]
+                ).Build();
+
+            Configuration = config;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddOptions();
+            services.Configure<ForecastingConfiguration>(Configuration.GetSection("Forecasting"));
+            services.AddSingleton(cfg => cfg.GetService<IOptions<ForecastingConfiguration>>().Value);
+            services.Configure<AzureActiveDirectoryConfiguration>(Configuration.GetSection("AzureAd"));
+            services.AddSingleton(cfg => cfg.GetService<IOptions<AzureActiveDirectoryConfiguration>>().Value);
+
+            var serviceProvider = services.BuildServiceProvider();
+            var azureActiveDirectoryConfiguration = serviceProvider.GetService<IOptions<AzureActiveDirectoryConfiguration>>();
+            var forecastingConfiguration = serviceProvider.GetService<IOptions<ForecastingConfiguration>>();
+
+            if (!Configuration["Environment"].Equals("LOCAL", StringComparison.CurrentCultureIgnoreCase))
```

Comment:
yeah - the problem is that its another field to set on the build templates for a parameter that we already use. I purposefully changed it to use this one as we use it for the development storage as well. Otherwise their are kinda two things doing the same task.